lib/src/network/protocol/http.rs

```diff
@@ -144,9 +144,14 @@ impl<Front:SocketHandler> Http<Front> {
   pub fn set_answer(&mut self, answer: DefaultAnswerStatus, buf: &[u8])  {
     self.front_buf.reset();
     self.back_buf.reset();
-    self.back_buf.write(buf);
-    self.back_buf.consume_parsed_data(buf.len());
-    self.back_buf.slice_output(buf.len());
+    match self.back_buf.write(buf) {
+      Ok(sz) => {
+        self.back_buf.consume_parsed_data(sz);
```

Comment:
please show an error here if `sz < buf.len()`. The call to `write()` can succeed but write only a part of what we want

Reply:
Done